pkg/activator/handler/handler.go

```diff
@@ -104,7 +139,15 @@ func (a *activationHandler) probeEndpoint(logger *zap.SugaredLogger, r *http.Req
 	var (
 		attempts int
 		st       = time.Now()
+		url      = target.String()
 	)
+	// This opputunistically caches the probes, so
+	// a few concurrent requests might result in concurrent probes
+	// but requests coming after won't.
+	if !a.cache.should(url) {
```

Comment:
Existential question: Why do we need to reprobe at all? Shouldn't we only need to probe as long as we didn't scale to 0 in the interim? And if we scaled to 0 and scale up again, we'd need to probe immediately again rather than wait for 5 seconds at worst. Currently the 5s wait won't harm, because scaling to 0 will take us longer than that anyway.

Reply:
I tried to keep it as small and as unintrusive as possible.

Reply:
This is part of the change id like to make with refactoring throttler/breaker - I think we should exhibit more typical backend health tracking for a LB which looks something like: periodically health check based on time and/or watch responses from forwarded requests to determine if a backend becomes unhealthy.